Library/CustomConfig.ts

```diff
@@ -0,0 +1,83 @@
+import { ICustomConfig } from "../Library/ICustomConfig";
+import Logging = require('./Logging');
+
+import fs = require("fs");
+
+const APPLICATION_INSIGHTS_CONFIG_PATH = "APPLICATION_INSIGHTS_CONFIG_PATH";
+
+// Azure Connection String
+export const ENV_connectionString = "APPLICATIONINSIGHTS_CONNECTION_STRING";
+// Native Metrics Opt Outs
+export const ENV_nativeMetricsDisablers = "APPLICATION_INSIGHTS_DISABLE_EXTENDED_METRIC";
+export const ENV_nativeMetricsDisableAll = "APPLICATION_INSIGHTS_DISABLE_ALL_EXTENDED_METRICS"
+export const ENV_http_proxy = "http_proxy";
+export const ENV_https_proxy = "https_proxy";
+export const ENV_noDiagnosticChannel = "APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL"
+export const ENV_noStatsbeat = "APPLICATION_INSIGHTS_NO_STATSBEAT";
+export const ENV_noHttpAgentKeepAlive = "APPLICATION_INSIGHTS_NO_HTTP_AGENT_KEEP_ALIVE";
+export const ENV_noPatchModules = "APPLICATION_INSIGHTS_NO_PATCH_MODULES";
+
+export class CustomConfig {
+    static _config: ICustomConfig;
+
+    public static generateConfigurationObject(): ICustomConfig {
+        if (this._config) {
+            return this._config;
+        }
+        this._config = {} as ICustomConfig;
+        this._config.connectionString = process.env[ENV_connectionString];
+        this._config.disableAllExtendedMetrics = !!process.env[ENV_nativeMetricsDisableAll];
+        this._config.extendedMetricDisablers = process.env[ENV_nativeMetricsDisablers];
+        this._config.proxyHttpUrl = process.env[ENV_http_proxy];
+        this._config.proxyHttpsUrl = process.env[ENV_https_proxy];
+        this._config.noDiagnosticChannel = !!process.env[ENV_noDiagnosticChannel];
+        this._config.disableStatsbeat = !!process.env[ENV_noStatsbeat];
+        this._config.noHttpAgentKeepAlive = !!process.env[ENV_noHttpAgentKeepAlive];
+        this._config.noPatchModules = process.env[ENV_noPatchModules] || "";
+        try {
+            const customConfigObj = JSON.parse(fs.readFileSync(process.env[APPLICATION_INSIGHTS_CONFIG_PATH] || "", "utf8"));
```

Comment:
You can include only this line in try block, unless and exception could be triggered on other code, looks weird as previous statements are not included